### DISCUSSION TITLE
populate() not working

### DISCUSSION THREAD

**Discussion by nikstaaar** [neuefische/questions#233] [Web] [answered]
hi, I am trying to build relations in mongoDB between my users and ingredients. I use populate for that but mongoDB doesn't really populate the different ingredients. I think the problem might be that the user is created with the gitHub provider and the mongoDB.  

my userModel:
<img width="988" alt="Screenshot 2022-10-10 at 02 04 00" src="https://user-images.githubusercontent.com/104061897/194819391-066b6b23-266f-4660-958e-df190bb8af4c.png">

the userModel for the provider
<img width="986" alt="Screenshot 2022-10-10 at 02 04 44" src="https://user-images.githubusercontent.com/104061897/194819406-ea74060b-73b0-4d71-b20e-cbeba5ac575f.png">

populate
<img width="974" alt="Screenshot 2022-10-10 at 02 03 38" src="https://user-images.githubusercontent.com/104061897/194819412-917c8f97-30f1-42d7-b866-b2349b4f39e2.png">

mongoDB
<img width="1258" alt="Screenshot 2022-10-10 at 02 02 41" src="https://user-images.githubusercontent.com/104061897/194819483-f4dbaac7-523c-4879-9252-3c0374ef6397.png">

<img width="1436" alt="Screenshot 2022-10-10 at 02 02 21" src="https://user-images.githubusercontent.com/104061897/194819493-511689a0-2231-4a9b-afca-adb2796fc6fe.png">

https://github.com/nikstaaar/capstone-project/tree/refacture

**Comment by mbosselmann** [comment]
Hi Niklas,
can you push your current code and link it here please?
Thank you ❤️

**Reply by nikstaaar** [reply]
it's not in my current code, because I removed all of that for now

**Reply by mbosselmann** [reply]
It is often a better idea to leave the code first for us to see which path you choose. Screenshots does not explain everything. And you told us in our daily that you refactored your whole app. It is helpful to see your current state. Please push this current state and link this to your question.

**Reply by nikstaaar** [reply]
could you please just come in zoom I could explain better there.

**Comment by ThomasKirst** [comment, marked as answer]
Hi Niklas,

your function which tries to enrich your user data with the selected ingredients, should read like this:

```js
export async function getUser(request, response) {
	try {
		const {userId} = request.query;
		// This line needs only one await and populates the ingredients directly – no exec() needed afterwards
		const user = await User.find({email: userId}).populate('ingredients');
		response.status(200).json(user);
	} catch (error) {
		console.error(error);
		response.status(404).json({error: 'Cannot get the User'});
	}
}
```

It is also important to mention that you include the 'ingredients' model within the user model.

```js
import mongoose, {Schema, model, models} from 'mongoose';
// 👇 THIS IS VERY IMPORTANT
import './ingredientsModel';

const userSchema = new Schema({
	name: String,
	userName: String,
	email: String,
	image: String,
	ingredients: [{type: mongoose.Schema.Types.ObjectId, ref: 'Ingredients'}],
	createdAt: String,
});
```
Please also note that your `ref`(reference) uses the same name as the model (`Ingredients`)
`ingredients: [{type: mongoose.Schema.Types.ObjectId, ref: 'Ingredients'}],` i => I

I hope this helps ;)
